pkg/controller/names.go

```diff
@@ -16,8 +16,15 @@
 
 package controller
 
-import "fmt"
+import (
+	"fmt"
+	"github.com/knative/eventing/pkg/utils"
+)
 
 func ServiceHostName(serviceName, namespace string) string {
-	return fmt.Sprintf("%s.%s.svc.cluster.local", serviceName, namespace)
+	clusterDomainName, err := utils.GetClusterDomainName()
+	if err != nil {
+		clusterDomainName = "cluster.local"
```

Comment:
the rational was that error handling might be different, if it is hidden in the func it might be less flexible for custom error handling.  does it make sense? wdyt